src/uwtools/tests/test_schemas.py

```diff
@@ -90,6 +96,117 @@ def test_schema_chgres_cube_run_dir(chgres_cube_prop):
     assert "88 is not of type 'string'" in errors(88)
 
 
+# esg-grid
+
+
+def test_schema_esg_grid():
+    config = {
+        "execution": {"executable": "esg_grid"},
+        "run_dir": "/tmp",
+    }
+    errors = schema_validator("esg-grid", "properties", "esg_grid")
+    # Basic correctness:
+    assert not errors(config)
+    # Some top-level keys are required:
+    for key in ("execution", "run_dir"):
+        assert f"'{key}' is a required property" in errors(with_del(config, key))
+    # Additional top-level keys are not allowed:
+    assert "Additional properties are not allowed" in errors({**config, "foo": "bar"})
+    # Additional MPI support is not allowed:
+    assert "Additional properties are not allowed ('mpicmd' was unexpected)" in errors(
+        {"execution": {"mpicmd": "srun"}}
+    )
+
+
+def test_schema_esg_grid_namelist(esg_grid_prop):
+    base_file = {"base_file": "/some/path"}
+    update_values = {
+        "update_values": {
+            "regional_grid_nml": {
+                "delx": 0.22,
+                "dely": 0.22,
+                "lx": -200,
+                "ly": -130,
+                "pazi": 0.0,
+                "plat": 45.5,
+                "plon": -100.5,
+            }
+        }
+    }
+    config = """
+        "regional_grid_nml": {
+            "delx": 0.22,
+            "dely": 0.22,
+            "lx": -200,
+            "ly": -130,
+            "pazi": 0.0,
+            "plat": 45.5,
+            "plon": -100.5,
+        }
+    """
+    cfg = yaml.safe_load(config)
+
+    errors = esg_grid_prop("namelist")
+    # Just base_file is ok:
+    assert not errors(base_file)
+    # base_file must be a string:
+    assert "88 is not of type 'string'" in errors({"base_file": 88})
+    # Just update_values is ok:
+    assert not errors(update_values)
+    # A combination of base_file and update_values is ok:
+    assert not errors({**base_file, **update_values})
+    # All key/value pairs in update_values must be present if no base_file is supplied:
+    assert "'dely' is a required property" in errors(
+        {"update_values": {"regional_grid_nml": {"delx": 0.11, "lx": -180, "plat": 38.0}}}
+    )
+
+    # Subsection of update_values is ok if base_file is supplied:
+    #  assert not errors(
+    #      {
+    #          **base_file,
+    #          "update_values": {"regional_grid_nml": {"delx": 0.11, "lx": -180, "plat": 38.0}},
+    #      }
+    #  )
+    # update_values values must be a number:
+    assert "'/some/str' is not of type 'number'" in errors(
+        {"update_values": {"regional_grid_nml": {"delx": "/some/str"}}}
+    )
+    # regional_grid_nml is required with update_values:
+    assert "'regional_grid_nml' is a required property" in errors({"update_values": {}})
+    # At least one is required:
+    assert "is not valid" in errors({})
+    # Just namelist file is not sufficient:
```

Comment:
@maddenp-noaa I'm working on this suggestion below

> I see now that it *is* ok for neither `base_file` or `update_values` to be provided: Just don't provide the `namelist` block at all! But if it _is_ provided, then at least one of `base_file` or `update_values` is required. That makes sense, and means that users can place their own namelist file in the run directory and not mention `namelist` at all in their UW YAML. Sorry for the confusion.

> It looks like our current schema demands at least one of `base_file` and `update_values`, so that edge case we discussed of a user placing a namelist file in the run directory before running the driver is probably not currently supported -- I mean, it'll work, but config validation will still complain if both `base_file` and `update_values` are missing. I think. It'd be good to make sure the schema unit tests check this and have a comment explaining why it is the way it is.

I added a YAML above called `config` and I am getting an error of:
```
E       TypeError: schema_validator.<locals>.<lambda>() got an unexpected keyword argument 'regional_grid_nml'
```

I thought I can record an exception here for a TypeError and then log "Just namelist file is not sufficient please supply `base_file` or `update_values`. But re-reading your comment maybe this isn't what you were looking for? Maybe we can discuss more on this together in UW Group.

Reply:
I don't think I completely understand what you were hoping to do, and see that `cfg` isn't actually used, but based on some other, more recent comments, I think the only thing we need to test -- if it's not already tested, and it might be -- is that *if* `namelist` is present (and it is currently optional), then at least one of `base_file` and/or `update_values` must also be present. That's what, if I'm not mistaken, the schema is written to express.